src/utils/dom-utils/inviteToOpenSauced.ts

```diff
@@ -0,0 +1,36 @@
+import { InviteToOpenSaucedButton } from "../../components/InviteToOpenSauced/InviteToOpenSaucedButton";
+import { InviteToOpenSaucedModal } from "../../components/InviteToOpenSauced/InviteToOpenSaucedModal";
+import { getTwitterUsername, getLinkedInUsername } from "../matchers";
+
+const injectViewOnOS = (username: string) => {
+  const emailAddress: string | undefined = (
+    document.querySelector(`a[href^="mailto:"]`) as HTMLAnchorElement
+  )?.href.substr(7);
+  const twitterUrl: string | undefined = (
+    document.querySelector(`a[href*="twitter.com"]`) as HTMLAnchorElement
+  )?.href;
+  const linkedInUrl: string | undefined = (
+    document.querySelector(`a[href*="linkedin.com"]`) as HTMLAnchorElement
+  )?.href;
+  if (!(emailAddress || twitterUrl || linkedInUrl)) return;
+
+  const twitterUsername = twitterUrl && getTwitterUsername(twitterUrl);
+  const linkedInUsername = linkedInUrl && getLinkedInUsername(linkedInUrl);
+  const inviteToOpenSaucedButton = InviteToOpenSaucedButton();
+  const inviteToOpenSaucedModal = InviteToOpenSaucedModal(username, {
+    emailAddress,
+    twitterUsername,
+    linkedInUsername,
+  });
+
+  inviteToOpenSaucedButton.onclick = () => {
```

Comment:
Maybe this should be a part of the component only, considering you added onclick events for other components in their files.

We might need to discuss what the exact difference between dom-utils and components will be. Like what goes where.

Reply:
I will leave this to you and @Anush008. We want to make sure the code and design are reusable and clear. We will identify best practices as we go along. 

related: We should introduce linting into the project too. 

we can copy from hot - https://github.com/open-sauced/hot/blob/beta/package.json#L50

Reply:
@diivi, as you suggested, I've refactored the modal component to receive an element that'll trigger it into view as an arg. Thereby removing setting the click listener in this file.
https://github.com/open-sauced/browser-extensions/pull/20/commits/7a91203671b2c2fd94806fe876cdc7c41a480696